lib/core/src/Cardano/Wallet/DB/Sqlite.hs

```diff
@@ -2165,13 +2219,24 @@ selectTxs = fmap concatUnzip . mapM select . chunksOf chunkSize
             [TxInputTxId <-. txids]
             [Asc TxInputTxId, Asc TxInputOrder]
 
+        collateralInputs <- fmap entityVal <$> selectList
+            [TxCollateralInputTxId <-. txids]
+            [Asc TxCollateralInputTxId, Asc TxCollateralInputOrder]
+
         resolvedInputs <- fmap toOutputMap $
             combineChunked inputs $ \inputsChunk ->
                 traverse readTxOutTokens . fmap entityVal =<<
                     selectList
                         [TxOutputTxId <-. (txInputSourceTxId <$> inputsChunk)]
                         [Asc TxOutputTxId, Asc TxOutputIndex]
 
+        resolvedCollateralInputs <- fmap toOutputMap $
+            combineChunked collateralInputs $ \collateralInputsChunk ->
+                traverse readTxOutTokens . fmap entityVal =<< selectList
```

Comment:
Note that we don't actually expect to read any tokens for collateral inputs.

We should consider simplifying this logic.